### ISSUE TITLE
rate limit / incremental download

### ISSUE THREAD

**Issue by allefeld** [javierarce/tumblr-liked-photos-export#24]
Does the program deal gracefully with tumblr's rate limit (1000 requests per hour, 5000 requests per day for new keys)? If not, can I restart the program so that checks whether a file already exists and doesn't download it again?

**Comment by javierarce** [comment]
No, it doesn't deal with the rate limit… but since the script is using the `before`  timestamp param, it could read a given `before`  timestamp and start downloading from there.

**Comment by JFischer84** [comment]
Sorry for the noobish question, but I am not used to ruby. How do I use the timestamp param to get more than 1000 requests?

**Comment by javierarce** [comment]
I'm going to close this issue, but I'll explain it a little bit more in case someone needs help:

If you need to resume the download of likes, you'll need to copy the `before` timestamp:

1. Print the timestamp modifying the [line 68](https://github.com/javierarce/tumblr-liked-photos-export/blob/master/export.rb#L68): 

```
    if @before
      puts "\033[32m#{before}\033[0m"
      response = HTTParty.get(@url + "&limit=#{limit}&before=#{@before}")
    else
      response = HTTParty.get(@url + "&limit=#{limit}")
    end
```

That will print a number in green. Then, if you need to restart the download, copy the last timestamp printed by the script and paste it on the [line 21](https://github.com/javierarce/tumblr-liked-photos-export/blob/master/export.rb#L21), like this:

```
 @before       = 1234567
```

Hopefully that should do the trick :)

**Comment by allefeld** [comment]
I understand, and thanks for the pointers.

Something I'd like to mention: I experimented a bit with the tumblr API myself trying to obtain my likes, and I found it extremely weird and unreliable. I went through the list first using the query/next value for the next request, and found it skips over likes. I then used the field `liked_timestamp` from the last returned post, that worked a little better. I experimented with the `limit` parameter, and found that for a small value, resulting in a lot of requests, at some point the API simply starts to return 0 posts, even though I now that the requested time point has many likes before it. Mind you, no error message, just an "OK" response containing zero posts. The number of posts returned via different methods amount to several numbers in the 5-thousands, even though there are supposed to be more than 9000 likes. Etc. etc.

I used your program, https://github.com/bbolli/tumblr-utils and its fork by aggroskater, and my own code, and I never arrive at the 9000+ likes, get different numbers of recovered posts each time and on different runs.

I'll be turning my back on tumblr soon, and just wanted to get my stuff out before the impending apocalypse. After the both social and technical blunders they commit, I have to say: Good riddance.

Sorry for venting. Thank you for your work!
